network/topics/msg_id.go

```diff
@@ -96,31 +99,57 @@ func (handler *msgIDHandler) MsgID(logger *zap.Logger) func(pmsg *ps_pb.Message)
 		if pmsg == nil {
 			return MsgIDEmptyMessage
 		}
-		//logger := logger.With()
-		if len(pmsg.GetData()) == 0 {
+
+		messageData := pmsg.GetData()
+		if len(messageData) == 0 {
 			return MsgIDEmptyMessage
 		}
+
 		pid, err := peer.IDFromBytes(pmsg.GetFrom())
 		if err != nil {
 			return MsgIDBadPeerID
 		}
-		mid := commons.MsgID()(pmsg.GetData())
+
+		currentEpoch := handler.networkConfig.Beacon.EstimatedCurrentEpoch()
+		if handler.networkConfig.RSAMessageFork(currentEpoch) {
+			decodedMsg, _, _, err := commons.DecodeSignedSSVMessage(messageData)
+			if err != nil {
+				logger.Debug("could not decode signed SSV message", zap.Error(err))
+			} else {
+				messageData = decodedMsg
+			}
+		}
+
+		mid := commons.MsgID()(messageData)
 		if len(mid) == 0 {
 			logger.Debug("could not create msg_id",
 				zap.ByteString("seq_no", pmsg.GetSeqno()),
 				fields.PeerID(pid))
 			return MsgIDError
 		}
+
 		handler.Add(mid, pid)
 		return mid
 	}
 }
 
 // GetPeers returns the peers that are related to the given msg
 func (handler *msgIDHandler) GetPeers(msg []byte) []peer.ID {
+	currentEpoch := handler.networkConfig.Beacon.EstimatedCurrentEpoch()
+	if handler.networkConfig.RSAMessageFork(currentEpoch) {
+		decodedMsg, _, _, err := commons.DecodeSignedSSVMessage(msg)
+		if err != nil {
+			// TODO: consider logging it
+		} else {
+			msg = decodedMsg
+		}
+	}
+
 	msgID := commons.MsgID()(msg)
```

Comment:
Since this code is used twice, maybe we can extract it to a method (e.g `msgIDFromPubSubMessage``)

Reply:
done